swift/Todo/Controller/TodoController.swift

```diff
@@ -0,0 +1,182 @@
+//
+//  TaskController.swift
+//  Todo
+//
+//  Created by Callum Birks on 20/02/2023.
+//  Copyright © 2023 Couchbase. All rights reserved.
+//
+
+import SwiftUI
+import CouchbaseLiteSwift
+
+// An interface between the UI and the DB class. A class that calls functions from this one should
+// implement TaskControllerDelegate (in order to present an error)
+
+class TodoController {
+    
+// - MARK: Queries
+    
+    public static func taskListsQuery() -> ObservableQuery {
+        guard let query = try? DB.shared.getTaskListsQuery()
+        else {
+            fatalError("Could not get task lists query")
+        }
+        return ObservableQuery(query)
+    }
+    
+    public static func tasksQuery(forList taskListID: String) -> ObservableQuery {
+        guard let query = try? DB.shared.getTasksQuery(taskListID: taskListID)
+        else {
+            fatalError("Could not get tasks query for task list ID: \(taskListID)")
+        }
+        return ObservableQuery(query)
+    }
+    
+    public static func usersQuery(forList taskListID: String) -> ObservableQuery {
+        guard let query = try? DB.shared.getSharedUsersQuery(taskList: getTaskListDoc(fromID: taskListID))
+        else {
+            fatalError("Could not get shared users for task list with ID: \(taskListID)")
+        }
+        return ObservableQuery(query)
+    }
+    
+// - MARK: Fetch documents by ID
+    
+    public static func getTaskListDoc(fromID taskListID: String) -> Document {
+        do {
+            guard let taskListDoc = try DB.shared.getTaskListByID(id: taskListID)
+            else {
+                fatalError("Couldn't fetch task list with ID: \(taskListID)")
```

Comment:
I think it's better to show an error dialog instead of crash. It's possible that the task list is deleted by the replicator.

Reply:
My new solution for this is to return nil if we fail to get the task list, and let the caller handle the error, as it can be called from places where we can't use TodoControllerDelegate.